repository/src/main/java/org/alfresco/repo/domain/schema/script/DeleteNotExistsV3Executor.java

```diff
@@ -0,0 +1,437 @@
+                // It can be further refined by being verified at each primary row processing.
+                while (hasResults && !isTimeoutExceeded())
+                {
+                    // Process batch
+                    primaryId = processPrimaryTableResultSet(primaryPrepStmt, secondaryPrepStmts, deletePrepStmt, deleteIds,
+                            primaryTableName, primaryColumnName, tableColumn);
+                    connection.commit();
+
+                    if (primaryId == null || pauseAndRecover)
+                    {
+                        break;
+                    }
+
+                    // Prepare for next batch
+                    primaryPrepStmt.setLong(1, primaryId + 1);
+                    primaryPrepStmt.setLong(2, tableUpperLimits[0]);
+
+                    hasResults = primaryPrepStmt.execute();
+                }
+
+                // Check if we have any more ids to delete
+                if (!deleteIds.isEmpty())
+                {
+                    deleteFromPrimaryTable(deletePrepStmt, deleteIds, primaryTableName);
+                    connection.commit();
+                }
+            }
+        }
+        finally
+        {
+            closeQuietly(deletePrepStmt);
+            closeQuietly(secondaryPrepStmts);
+            closeQuietly(primaryPrepStmt);
+
+            closeQuietly(connection);
+        }
+
+        if (pauseAndRecover)
+        {
+            pauseAndRecoverJob(dataSource);
+            logger.info("Resuming the job on primary table " + primaryTableName + " picking up after id " + primaryId);
+            processBatch(primaryTableName, primaryColumnName, primaryWhereClause, primaryId, tableColumn, tableUpperLimits,
+                    optionalWhereClauses);
+        }
+    }
+
+    @Override
+    protected String createPreparedSelectStatement(String tableName, String columnName, String whereClause)
+    {
+        StringBuilder sqlBuilder = new StringBuilder("SELECT " + columnName + " FROM " + tableName + " WHERE ");
+
+        if (whereClause != null && !whereClause.isEmpty())
+        {
+            sqlBuilder.append(whereClause + " AND ");
+        }
+
+        sqlBuilder.append(
+                columnName + " >= ? AND " + columnName + " <= ? GROUP BY " + columnName + " ORDER BY " + columnName + " ASC ");
+
+        if (dialect instanceof MySQLInnoDBDialect)
+        {
+            sqlBuilder.append(" LIMIT " + batchSize);
+        }
+        else
+        {
+            sqlBuilder.append(" OFFSET 0 ROWS FETCH FIRST " + batchSize + " ROWS ONLY");
+        }
+
+        return sqlBuilder.toString();
+    }
+
+    @Override
+    protected Long processPrimaryTableResultSet(PreparedStatement primaryPrepStmt, PreparedStatement[] secondaryPrepStmts,
+            PreparedStatement deletePrepStmt, Set<Long> deleteIds, String primaryTableName, String primaryColumnName,
+            Pair<String, String>[] tableColumn) throws SQLException
+    {
+        int rowsProcessed = 0;
```

Comment:
I don't think we need to initialise this here - it always gets overridden below.